segmentstore/storage/impl/src/test/java/io/pravega/segmentstore/storage/impl/extendeds3/ExtendedS3StorageTest.java

```diff
@@ -73,6 +77,31 @@ public void tearDown() throws Exception {
         s3Proxy.stop();
     }
 
+    //region If-none-match test
+    /**
+     * Tests the create() method with if-none-match set.
+     */
+    @Test
+    public void testCreateIfNoneMatch() {
+        adapterConfig = ExtendedS3StorageConfig.builder()
+                                               .with(ExtendedS3StorageConfig.BUCKET, adapterConfig.getBucket())
+                                               .with(ExtendedS3StorageConfig.ACCESS_KEY_ID, "x")
+                                               .with(ExtendedS3StorageConfig.SECRET_KEY, "x")
+                                               .with(ExtendedS3StorageConfig.ROOT, "test")
+                                               .with(ExtendedS3StorageConfig.URI, endpoint)
+                                               .with(ExtendedS3StorageConfig.IFNONEMATCHWORKS, true)
+                                               .build();
+        String segmentName = "foo_open";
+        try (Storage s = createStorage()) {
+            s.initialize(DEFAULT_EPOCH);
+            s.create(segmentName, null).join();
+            assertThrows("create() did not throw for existing StreamSegment.",
```

Comment:
As per our offline discussion, this test case is not really verifying that the option works as out unit tests do not exercise a real S3-like store. Instead, this is just checking that the new code path is not breaking anything else.